src/required_resources.h

```diff
@@ -150,16 +150,26 @@ class RequiredResources {
   std::vector<
       std::tuple<FormatString, std::vector<Field>, PrintfSeverity, SourceInfo>>
       printf_args;
+  std::unordered_map<ast::Call* , size_t> printf_args_id_map;
```

Comment:
Could we leave a comment what these `*_args` and `*_args_id_map` collections are for?